components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/JSONInputReader.java

```
@@ -0,0 +1,91 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.mediator.datamapper.engine.input.readers;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.mediator.datamapper.engine.core.exceptions.JSException;
+import org.wso2.carbon.mediator.datamapper.engine.core.exceptions.ReaderException;
+import org.wso2.carbon.mediator.datamapper.engine.core.exceptions.SchemaException;
+import org.wso2.carbon.mediator.datamapper.engine.core.schemas.Schema;
+import org.wso2.carbon.mediator.datamapper.engine.input.InputBuilder;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+/**
+ * This class is capable of parsing XML through AXIOMS for the InputStream and build the respective JSON message
+ */
+public class JSONInputReader implements InputReader {
+
+    private static final Log log = LogFactory.getLog(JSONInputReader.class);
+
+    /**
+     * Constructor
+     *
+     * @throws IOException
+     */
+    public JSONInputReader() throws IOException {
+    }
+
+    /**
+     * Read, parse the XML and notify with the output JSON message
+     *
+     * @param input          XML message InputStream
+     * @param inputSchema    Schema of the input message
+     * @param messageBuilder Reference of the InputXMLMessageBuilder
+     * @throws ReaderException Exceptions in the parsing stage
+     */
+    @Override
+    public void read(InputStream input, Schema inputSchema, InputBuilder messageBuilder) throws ReaderException {
+        String inputJSONMessage;
+        try {
+            inputJSONMessage = readFromInputStream(input);
+            messageBuilder.notifyWithResult(inputJSONMessage);
+        } catch (IOException e) {
```

we can catch all exceptions in one catch clause. No need to modify the message since e.getMessage() will contain all info